Lib/email/_header_value_parser.py

```diff
@@ -2404,11 +2404,13 @@ def get_section(value):
                                         value))
     section.append(ValueTerminal('*', 'section-marker'))
     value = value[1:]
-    if not value or not value[0].isdigit():
+    # We don't use str.isdigit because only 0-9 are accepted, not super-script
+    # and other types of digits.
+    if not value or not '0' <= value[0] <= '9':
         raise errors.HeaderParseError("Expected section number but "
                                       "found {}".format(value))
     digits = ''
-    while value and value[0].isdigit():
+    while value and '0' <= value[0] <= '9':
```

Comment:
```suggestion
    while value and ('0' <= value[0] <= '9'):
```

It will a bit clearer. Or you can still use a separate function to make it even cleareer. The bottleneck won't be the function call IMO.

Reply:
I did that, but not the separate function. It was my understanding that @StanFromIreland was leaning towards not having an inner function

Reply:
This is fine, I was against the function to check if it is in a dictionary.

Reply:
Thank you! Just moved it to a separate function for extra-clarity